### ISSUE TITLE
STUPID ERROR ADDED TO ModuleTrait.php

### ISSUE THREAD

**Issue by Ziggizag** [dektrium/yii2-user#1008]
There was modification introduced in ModuleTrait.php adding this code:

```
    /**
     * @return string
     */
    public static function getDb()
    {
        return \Yii::$app->getModule('user')->getDb();
    }
```

This effectively breaks the application in case any relation to User is used in console mode. Please - undo!!!

**Comment by SamMousa** [comment]
@thiagotalma feel free to close this issue.

As explained here: https://github.com/dektrium/yii2-user/commit/4b06f2b057b488091ae91f77e749890e189c7376#commitcomment-27019680

This is a configuration issue that can easily be resolved.

**Comment by Ziggizag** [comment]
@SamMousa 

You have added changes to this fundamental module what effectively rendered production application broken since composer update. There was neither warning nor changelog info on the subject. Just "BUM".

Sure - you may complain people get upset on that. There is no liability on open source...

**Comment by SamMousa** [comment]
> You have added changes to this fundamental module what effectively rendered production application broken since composer update. There was neither warning nor changelog info on the subject. Just "BUM".

That's what happen if you couple your application to implementation details, the generic documentation of this module tells you to configure it, the fact that you did not means you did not follow instructions.

> Sure - you may complain people get upset on that. There is no liability on open source...

Regardless of whether you are upset or not, it doesn't give you the right to show disrespect.

**Comment by thiagotalma** [comment]
@Ziggizag you may be helpful by sending a PR with the fix for the problem.
Any further discussion is unnecessary.

**Comment by SamMousa** [comment]
Okay, @Ziggizag apologized so we're good for now.

Some explanation regarding your issue:
1. The whole module should be seen as unit of coupling; that means that you MUST not assume you can use parts of the module without setting it up correctly.

2. The models in this module always have depended on the `Module` to be configured, you just got lucky and didn't call some methods on the model that caused you to think your configuration was ok. For example, consider these lines:

- https://github.com/dektrium/yii2-user/blob/master/models/User.php#L121
- https://github.com/dektrium/yii2-user/blob/master/models/User.php#L134
- https://github.com/dektrium/yii2-user/blob/master/models/User.php#L151

And there's more.


So in conclusion it is not the changes that create strong(er) coupling, it is your application. The best way forward in my opinion is to simply configure the module in your console application.

Good luck.
